File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    const store = {}
    const N = nums.length / 2
    for (const num of nums){
        store[num] += (store[num] ?? 0) + 1
    }
    const keys = Object.keys(store)
    return N < keys.length ? N : keys.length
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
using namespace std;
#include <unordered_map>

int solution(vector<int> nums)
{
    unordered_map<int,int> mp;
    // N / 2개 or N 개 nums길이보다 N / 2가 크면 nums길이개 작으면 N / 2개
    for (int num : nums){
        mp[num]++;
    }
    return min(mp.size(),nums.size() / 2);
}